zipline/data/data_portal.py

```diff
@@ -696,6 +766,25 @@ def get_history_window(self, assets, end_dt, bar_count, frequency, field,
 
         # forward-fill if needed
         if field == "price" and ffill:
+            assets_with_nan_index = np.where(pd.isnull(df.iloc[0, :]))[0]
+            assets_with_leading_nan = [
```

Comment:
This is replicating the behavior of masking with a boolean array.  Should we make `assets` a numpy array up above to make this cleaner?